tests/test_integration_scenarios.py

```
@@ -0,0 +1,512 @@
+class TestBoundaryConditions:
+    """Test boundary conditions and edge cases."""
+
+    def test_zero_check_interval_validation(self, mock_config, mock_state, mock_logger):
+        """Test that zero check interval is corrected."""
+        from gengowatcher.watcher import GengoWatcher
+
+        mock_config.get.side_effect = lambda s, k, **kw: {
+            ("Watcher", "check_interval"): 0
+        }.get((s, k), kw.get("fallback", 60))
+
+        watcher = GengoWatcher(mock_config, mock_state, mock_logger)
```

Ruff (0.14.14)

Local variable `watcher` is assigned to but never used

Remove assignment to unused variable `watcher`

(F841)